types/schemas/prototypeApplication/data/ApplicationData.ts

```diff
@@ -3,30 +3,213 @@ import {PrimaryApplicationType} from '../enums/ApplicationType';
+  };
+  reduction: {
+    alternative: boolean;
+    parishCouncil: boolean;
+    sports: boolean;
+  };
+  reference?: {
+    /**
+     * @description GOV.UK Pay payment reference number
+     */
+    govPay: string; // @todo require when payable > 0
+  };
+}
+
+/**
+ * @description Declarations about the accuracy of this application and any personal connections to the receiving authority
+ */
+export interface ApplicationDeclaration {
+  accurate: boolean;
+  connection: {
+    value:
+      | 'employee'
+      | 'relation.employee'
+      | 'electedMember'
+      | 'relation.electedMember'
+      | 'none';
+    description?: string;
+  };
+}
+
+/**
+ * @description Details of the pre-application preceeding this application, if applicable
+ */
+export interface PreApplication {
+  reference: string;
+  date: Date;
+  officer: string;
+  summary: string;
+}
+
+/**
+ * @description Details of the planning application linked to this application, if applicable
+ */
+export interface PlanningApplication {
+  reference: string;
+  date: Date;
+  localPlanningAuthority: string;
+}
+
+/**
+ * @description Details about the Community Infrastructure Levy planning charge, if applicable
  */
-export interface PPApplicationData extends ApplicationDataBase {
-  ppSpecificProperty: number;
+export interface CommunityInfrastructureLevy {
+  result:
+    | 'exempt.annexe'
+    | 'exempt.extension'
+    | 'exempt.selfBuild'
+    | 'liable'
+    | 'relief.charity'
+    | 'relief.socialHousing';
+}
+
+export interface LeadDeveloper {
+  type: 'ukCompany' | 'overseasCompany' | 'none';
+  company?: {
+    name: string;
+    registrationNumber: string;
+  };
 }
 
+/**
+ * @description Application details for project sites within the Greater London Authority (GLA) area
+ */
+export interface LondonApplicationData extends UKApplicationData {
+  vacantBuildingCredit?: boolean;
+  leadDeveloper?: LeadDeveloper;
+}
+
+/**
+ * @description ApplicationData required for fee carrying application types
+ */
+export type FeeCarryingApplicationData = ApplicationDataBase & {
+  fee: ApplicationFee;
+};
+
+/**
+ * @description ApplicationData required for application types that do not have a fee
+ */
+export type NonFeeCarryingApplicationData = ApplicationDataBase & {
+  fee: ApplicationFeeNotApplicable;
+};
+
+/**
+ * @description Specific ApplicationData required for "Planning Permission" applications
+ */
+export type PPApplicationData = FeeCarryingApplicationData & {
+  CIL?: CommunityInfrastructureLevy;
+};
+
 /**
  * TypeMap of PrimaryApplicationTypes to their specific ApplicationData models
  */
 export interface ApplicationDataVariants {
-  wtt: WTTApplicationData;
+  ldc: FeeCarryingApplicationData;
+  pa: FeeCarryingApplicationData;
   pp: PPApplicationData;
+  listed: NonFeeCarryingApplicationData;
+  landDrainageConsent: FeeCarryingApplicationData;
```

Comment:
Again, nice migration win in here of now explicitly defining which application types are "fee carrying" and which are not. There's a big semantic difference between `fee: 0` & `fee: "notApplicable"` that is now better communicated, no longer confusable based on a generic type !